jitsi-meet: stable/jitsi-meet_5390 -> stable/jitsi-meet_5963

illustris opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/120583 is out of date and appears to have stalled. Motivation remains the same, "Jitsi has gotten more functionality visible in the UI in the most recent version over the one that is currently in nixpkgs, there is also a [variety of fixes](https://github.com/jitsi/jitsi-meet-release-notes/blob/master/CHANGELOG-WEB.md)."

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

This PR is a WIP. The newer versions of jitsi-meet and jitsi-videobridge work fine, but updating Jicofo breaks Jitsi.

illustris committed:
jitsi-meet: stable/jitsi-meet_5390 -> stable/jitsi-meet_5963

illustris committed:
jitsi-videobridge: 2.1-416 -> 2.1-508

illustris committed:
jicofo: 1.0-690 -> 1.0-756

illustris commented:
Tagging maintainers
@mmilata @petabyteboy @ryantm

Does anyone know why upgrading jicofo causes issues? It might also be just on my setup, is anyone able to reproduce this issue?

illustris committed:
nixos/jitsi-meet: Update jitsi prosody configs

Changes made as per https://github.com/jitsi/jitsi-meet/commit/b6f7f8fba7ba81c5a0ab5477a1d103b7279bf38f

illustris commented:
https://github.com/NixOS/nixpkgs/pull/127236/commits/0d245614871bec72af16239464cff622671d73fc should account for the changes made in https://github.com/jitsi/jitsi-meet/commit/b6f7f8fba7ba81c5a0ab5477a1d103b7279bf38f.

There is one last issue though. The [prosody-plugins](https://github.com/jitsi/jitsi-meet/tree/stable/jitsi-meet_5963/resources/prosody-plugins) folder does not exist in the Jitsi web deb file or "https://download.jitsi.org/jitsi-meet/src/jitsi-meet-${version}.tar.bz2". The plugins folder must be included in the jitsi-meet service definition as `services.prosody.extraPluginPaths = [ "/path/to/prosody-plugins" ];`. I tested this in my deployment by just copying the folder locally and adding `extraPluginPaths = [ ./prosody-plugins ];`. Should this folder just be fetched from github, or is there a better way?

illustris committed:
jitsi-meet-prosody: init at 1.0.5056

illustris committed:
nixos/jitsi-meet: include jitsi prosody plugins in prosody extraPluginPaths

illustris commented:
Tested, works. @mmilata @petabyteboy @ryantm please review.

SuperSandro2000 commented on pkgs/misc/jitsi-meet-prosody/default.nix:
```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv, fetchurl, dpkg, nixosTests }:
+
+let
+  pname = "jitsi-meet-prosody";
+  version = "1.0.5056";
+  src = fetchurl {
+    url = "https://download.jitsi.org/stable/${pname}_${version}-1_all.deb";
+    sha256 = "06qxa9h2ry92xrk2jklp76nv3sl8nvykdvsqmhn33lz6q6vmw2xr";
+  };
+in
+stdenv.mkDerivation {
+  inherit pname version src;
```
```suggestion
stdenv.mkDerivation rec {
  pname = "jitsi-meet-prosody";
  version = "1.0.5056";

  src = fetchurl {
    url = "https://download.jitsi.org/stable/${pname}_${version}-1_all.deb";
    sha256 = "06qxa9h2ry92xrk2jklp76nv3sl8nvykdvsqmhn33lz6q6vmw2xr";
  };
```
Don't be afraid of rec.

illustris replied:
I keep forgetting this is a thing... Updated

petabyteboy commented:
Works fine in manual testing, but the NixOS tests time out?

illustris committed:
nixos/jitsi-meet: update nixos tests

- remove check for `connected .JID: focus@auth.server` because
	- log format was changed in https://github.com/jitsi/jicofo/commit/c1945ea6cb3dd7c815e963768abc9a3fab57101e
	- connection.getUser() in jicofo also appears to be broken, returning null instead of username
	- testing for this log line shouldn't be necessary, as we also test for "Authenticated as focus@auth.server"

- remove check for `External component successfully authenticated` because
	- [JVB no longer uses component](https://community.jitsi.org/t/jvb-not-connecting/91157/2)

- increase VM memory

illustris commented:
@petabyteboy I've updated the tests

petabyteboy commented:
In local testing the test runs out of memory, but adding `virtualisation.memorySize = 512;` to the server block fixes that.

illustris commented:
Huh... yes, looks like it fails sometimes. It worked the first time, failed the second time and worked again the third time. Updated test configs.

petabyteboy commented:
Please add a `nixos/jitsi-meet: ` prefix to the commit message of 0d245614871bec72af16239464cff622671d73fc

illustris commented:
Updated commit message prefixes

petabyteboy commented:
Thanks a lot!
